### PR TITLE
Enforces conversion of torque layers and removes warning dialog when adding another torque layer.

### LINKED ISSUE #2584: Two torque layers can be created
1. Import populated places table
2. Create a visualization
3. Assign heatmap (static) wizard
4. Add another layer (as an empty one)
5. In my case, I added some manual points as the table was empty.
6. Apply a Torque map. Below the wizard, you get the error "Sorry, but for the moment one Torque layer is allowed". **The Torque layer is there!**
7. Go to the heatmap layer, enable "animated".

Refresh the page:

image

Result:
https://team.cartodb.com/u/iriberri/viz/2c9ab0b8-c19b-11e4-82e0-0e018d66dc29/embed_map

cc @javisantana  @fdansv

### PR DESCRIPTION
**Pull request by alonsogarciapablo**
This PR fixes #2584 for happening. Still, it's possible to have multiple torque layers in a map by selecting the torque wizard multiple times. I'm working on this next.

@javisantana can you please take a look?

### CODE REVIEW

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2703/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2790/))

**Comment by alonsogarciapablo** [comment]
@javisantana. when you have a second, I would love to hear what you think of this. Thx!

File: lib/assets/javascripts/cartodb/models/map.js [hunk 1]
```diff
@@ -304,17 +304,6 @@ cdb.admin.Layers = cdb.geo.Layers.extend({
   },
 
   add: function(models, options) {
```
Comment (javisantana) [inline lib/assets/javascripts/cartodb/models/map.js]: this overrride is not need anymore, right?

**Comment by javisantana** [comment]
apart of the comment I don't know if there is a test for that (or is easy to add)

:es:

**Comment by iriberri** [comment]
subscribing, I can test tomorrow if needed

**Comment by alonsogarciapablo** [comment]
thanks @iriberri

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2856/))

**Comment by iriberri** [comment]
Let me know when this is deployed :)

On Tuesday, April 7, 2015, Cartofante <notifications@github.com> wrote:

> Frontend tests were OK [image: :+1:] (details
> <http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2856/>)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/CartoDB/cartodb/pull/2981#issuecomment-90656890>.
>


-- 

Carla Iriberri
*CartoDB*

**Comment by javisantana** [comment]
ping, could we deploy this @alonsogarciapablo  ?

**Comment by iriberri** [comment]
I'm testing. Give me some minutes!

**Comment by alonsogarciapablo** [comment]
Thanks @iriberri! We found an issue. [We are trying to force the activation of the polygon wizard](https://github.com/CartoDB/cartodb/blob/dc3cbba77d26c0acbcf5d70476199df76aaa5d55/lib/assets/javascripts/cartodb/models/map.js#L775). But [the wizard is not being activated because the table has an empty `geometry_types`](https://github.com/CartoDB/cartodb/blob/dc3cbba77d26c0acbcf5d70476199df76aaa5d55/lib/assets/javascripts/cartodb/models/wizard.js#L347-L351) (`table.geomColumnTypes()` is returning `[]`). And the thing is that [we're explicitly "renaming" `geometry_types` to `stats_geometry_types` when we parse the table metadata](https://github.com/CartoDB/cartodb/blob/dc3cbba77d26c0acbcf5d70476199df76aaa5d55/lib/assets/javascripts/cartodb/models/table.js#L133-L137).

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2892/))

**Comment by alonsogarciapablo** [comment]
@javisantana please check out [this last commit](https://github.com/CartoDB/cartodb/commit/16dc991036201d0e82809f4223ae3c3c2eb6cd25). It seems to be working now :)

**Comment by iriberri** [comment]
I can test tomorrow first hour! :) or maybe even this night, just lmk the staging :)

**Comment by alonsogarciapablo** [comment]
@iriberri great! I've chatted you the details.

**Comment by alonsogarciapablo** [comment]
@iriberri thanks for your feedback! We're now correctly doing the enforcement when we add a layer, but it's still posible to have several torque layers when using the wizards.

**Comment by alonsogarciapablo** [comment]
@javisantana In order to move forward and close WIP, and since part of the issue is solved (removing the warning dialog and preventing a visualization with multiple torque layers from even being added to a visualization), I would just merge this PR and continue working/discussing and preventing multiple layers to be selected in the wizards in a separate issue.

**Pull request merged**